planner/core/exhaust_physical_plans.go

```diff
@@ -1574,7 +1574,18 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		return nil, false
 	}
 	joins := make([]PhysicalPlan, 0, 8)
-	if p.ctx.GetSessionVars().AllowBCJ {
+	if p.ctx.GetSessionVars().AllowMPPExecution {
+		if p.ctx.GetSessionVars().AllowBCJ {
+			mppJoins := p.tryToGetMppHashJoin(prop, true)
+			if (p.preferJoinType & preferBCJoin) > 0 {
+				return mppJoins, true
+			}
+			joins = append(joins, mppJoins...)
+		} else {
```

Comment:
delete this else, allow bcj bust not disable MPP hash join.

Reply:
we select bcj or hash join on the basis of variable but not cost.

Reply:
add a TODO, decide by cost later?